pom.xml

```diff
@@ -420,14 +430,64 @@
   </repositories>
 
   <profiles>
+
+    <profile>	<!-- run SCA for master branch: -Dbranch=master -->
+      <id>static-code-analysis</id>
+      <activation>
+        <property>
```

Comment:
Should we remove the `branch=main` property here and enable devs to run this on any branch (especially code coverage report in local laptop/desktop dev environments, without changing pom)?

We can always do `[ "${{ github.ref_name }}" = "main" ] && mvn -P static-code-analysis clean package` to activate this profile for the CI.

Reply:
agree, took a spin to the CI workflow to work with PRs and main pushes. With PRs, we will also pass specific information that sonar can use to build the analysis for the changes